compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -501,10 +505,31 @@ private void copyJar(ZipArchiveOutputStream outStream, JarLibrary jarLibrary,
 
         // Transfers selected entries from this zip file to the output stream, while preserving its compression and
         // all the other original attributes.
-        zipFile.copyRawEntries(outStream, predicate);
+        zipFile.copyOptimisedRawEntries(outStream, predicate);
+//        zipFile.copyRawEntries(outStream, predicate);
         zipFile.close();
     }
 
+    private static HashMap<String, HashSet<String>> getDeadFunctionList(String pkgName, CompilerContext compilerContext) {
+        org.wso2.ballerinalang.compiler.PackageCache packageCache = org.wso2.ballerinalang.compiler.PackageCache.getInstance(compilerContext);
+
+        BPackageSymbol pkgSymbol = packageCache.getSymbol(pkgName);
+        int majorVersion = packageCache.getSymbol(pkgName).descriptor.version().value().major();
+
+        HashMap<String, HashSet<String>> classWiseDeadFunctionMap = new HashMap<>();
+
+        for (BInvokableSymbol deadFunction:pkgSymbol.deadFunctions) {
+            if (deadFunction.source != null) {
+                String className = deadFunction.source.replace(".bal", ".class");
+                String classLocationName = pkgName + "/" + majorVersion + "/" + className;
```

Comment:
This `\` may not work in windows. Please check

Reply:
It is working for windows. But haven't checked with Linux or Mac OS. But I think because this `classLocationName` is used later the by Apache `ZipFile` library, they handle it accordingly I guess. Will update this thread if we get any issues later on.